prow/spyglass/lenses/links/links.go

```diff
@@ -127,24 +132,54 @@ func toLink(jobPath string, content []byte, spyglassConfig config.Spyglass) link
 	}
 }
 
+func parseLinkFile(jobPath string, content []byte, spyglassConfig config.Spyglass) (linkGroup, error) {
+	switch extension := filepath.Ext(jobPath); extension {
+	case ".txt":
+		// Wrap a single link inside a linkGroup.
+		wrappedLink := toLink(jobPath, content, spyglassConfig)
+		return linkGroup{
+			Title: wrappedLink.Name,
+			Links: []link{
+				wrappedLink,
+			},
+		}, nil
+	case ".json":
+		group := linkGroup{}
+		if err := json.Unmarshal(content, &group); err != nil {
+			return group, err
+		}
+		for i := range group.Links {
+			group.Links[i].Link = clickableLink(group.Links[i].URL, spyglassConfig)
+		}
+		return group, nil
+	}
+	return linkGroup{}, fmt.Errorf("unrecognized file extension for artifact %v", jobPath)
```

Comment:
I don't really like the idea of adding additional runtime validation. The types of files that can be handled by the lens are already controlled via the configuration in `config.yaml`. If it is important to restrict the file extensions that this lens can handle for some reason then that should be enforced with config validation of the regular expression in `config.yaml` at presubmit time rather than allowing problematic config to merge that can lead to a runtime error.
I don't think we should enforce this constraint on the file extensions in code anyways though. It is technically a breaking change and the same constraint can be enforced optionally and more transparently via the configured regex if that is desired.

Reply:
SGTM, adjusted the code accordingly. Thanks for the follow-up justification.